Text edits with the same start position are applied in reverse

### Summary

Text edits returned from the LSP that have the same start position are being applied in reverse.
According to the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textEditArray), text edits with the same start position should be applied in the order received.

Recently, the language server client included with Neovim has fixed this bug (https://github.com/neovim/neovim/issues/29202)

Accordingly, using some servers (e.g. [efm-langserver](https://github.com/mattn/efm-langserver)) with the client included with Neovim, the formatted lines of the result may be reversed.
So I tried to ask the server to fix it (https://github.com/mattn/efm-langserver/issues/281), but the fix causes problems with other (non-specified) clients such as Helix.
Therefore, I would like you to implement a fix in Helix that follows the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textEditArray), as well as Neovim.

### Reproduction Steps

1. Install efm-langserver with patch to follow the LSP spec.

```console
$ git clone --branch fix-281 https://github.com/kyoh86/efm-langserver.git
$ cd efm-langserver
$ go install .
```

2. Set the path

```console
$ export PATH="$(go env GOPATH)/bin:$PATH"
```

3. Configure the Helix

helix/languages.toml

```
[language-server.efm]
command = "efm-langserver"

[[language]]
name = "go"
language-servers = [
  { name = "efm", only-features = [ "diagnostics", "format" ] },
]
```

4. Configure the efm-langserver

```
version: 2

tools:
  echo-fix: &echo-fix
    format-command: zsh -c 'print "foo\nbar\nbaz"'
    format-stdin: true

languages:
  json:
    - <<: *echo-fix
```

5. Open helix

```console
$ hx foo.json
```

6. Call `:format`


Expected result: 

```
foo
bar
baz
```

But actual result:

```
baz
bar
foo
```

If you install the efm-langserver without the patch, this doesn't happen.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-07-15T14:46:45.102 helix_lsp::transport [ERROR] efm err <- "2024/07/15 14:46:45 jsonrpc2 handler: sending response 2: jsonrpc2: connection is closed\n"
2024-07-15T14:46:45.104 helix_lsp::transport [ERROR] efm err <- "2024/07/15 14:46:45 efm-langserver: connections closed\n"
2024-07-15T14:46:45.102 helix_lsp::client [WARN] language server failed to terminate gracefully - server closed the stream
2024-07-15T14:46:45.104 helix_lsp::transport [ERROR] efm err: <- StreamClosed
2024-07-15T14:46:45.104 helix_lsp::transport [ERROR] efm err: <- IO(Os { code: 32, kind: BrokenPipe, message: "Broken pipe" })
```

</details>


### Platform

Linux

### Terminal Emulator

WezTerm

### Installation Method

releases page

### Helix Version

helix 24.7 (079f5442)

--- a/helix-lsp/src/lib.rs
+++ b/helix-lsp/src/lib.rs
@@ -503,7 +503,7 @@ pub mod util {
     ) -> Transaction {
         // Sort edits by start range, since some LSPs (Omnisharp) send them
         // in reverse order.
-        edits.sort_unstable_by_key(|edit| edit.range.start);
+        edits.sort_by_key(|edit| edit.range.start);
 
         // Generate a diff if the edit is a full document replacement.
         #[allow(clippy::collapsible_if)]

--- a/helix-term/src/commands/lsp.rs
+++ b/helix-term/src/commands/lsp.rs
@@ -1360,7 +1360,7 @@ fn compute_inlay_hints_for_view(
 
             // Most language servers will already send them sorted but ensure this is the case to
             // avoid errors on our end.
-            hints.sort_unstable_by_key(|inlay_hint| inlay_hint.position);
+            hints.sort_by_key(|inlay_hint| inlay_hint.position);
 
             let mut padding_before_inlay_hints = Vec::new();
             let mut type_inlay_hints = Vec::new();

